packages/cli/src/services/FileCommandLoader.ts

```diff
@@ -0,0 +1,215 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { promises as fs } from 'fs';
+import path from 'path';
+import toml from '@iarna/toml';
+import {
+  Config,
+  getProjectCommandsDir,
+  getUserCommandsDir,
+} from '@google/gemini-cli-core';
+import { ICommandLoader } from './types.js';
+import { CommandKind, SlashCommand } from '../ui/commands/types.js';
+
+/**
+ * Defines the structure of a valid command definition within a .toml file.
+ * This is the raw shape of the data after being parsed from TOML.
+ */
+interface TomlCommandDef {
+  prompt: string;
+  description?: string;
+}
+
+/**
+ * A type guard to safely validate that a parsed TOML object conforms to our
+ * required command definition structure.
+ *
+ * @param obj The unknown object parsed from a TOML file.
+ * @returns True if the object is a valid TomlCommandDef, false otherwise.
+ */
+function isTomlCommandDef(obj: unknown): obj is TomlCommandDef {
```

Comment:
Oh this is interesting. SO any additional property etc. we'll need to add to this method for validation? Is there no better way?

Reply:
Beautiful, that's a great choice